### PR TITLE
Add some small polish to the Kubernetes deployment process

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Review by iocanel — APPROVED** [review]
LGTM

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void deploy(KubernetesClientBuildItem kubernetesClient,
 
         if (!containerImage.isPresent()) {
             LOG.warn(
-                    "A Kubernetes deployment was requested by no extension was found to build a container image. Consider adding one of following extensions: \"quarkus-container-image-jib\", \"quarkus-container-image-docker\" or \"quarkus-container-image-s2i\",");
+                    "A Kubernetes deployment was requested but no extension was found to build a container image. Consider adding one of following extensions: \"quarkus-container-image-jib\", \"quarkus-container-image-docker\" or \"quarkus-container-image-s2i\",");
```
Comment (gsmet) [inline extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java]:
```suggestion
                    "A Kubernetes deployment was requested but no extension was found to build a container image. Consider adding one of following extensions: \"quarkus-container-image-jib\", \"quarkus-container-image-docker\" or \"quarkus-container-image-s2i\".");
```

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployerPrerequisite.java [hunk 2]
```diff
@@ -21,6 +26,13 @@ public void prepare(ContainerImageInfoBuildItem containerImage) {
         buildRequestProducer.produce(new ContainerImageBuildRequestBuildItem());
         if (containerImage.getRegistry().isPresent()) {
             pushRequestProducer.produce(new ContainerImagePushRequestBuildItem());
+        } else {
+            LOGGER.warn(
+                    "A Kubernetes deployment was requested, but the container image to be built will not be pushed to any registry"
+                            +
+                            " because \"quarkus.container-image.registry\" has not been set. The Kubernetes deployment will only work properly"
+                            +
+                            " if the cluster is using the local docker daemon.");
```
Comment (gsmet) [inline extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployerPrerequisite.java]:
```suggestion
                            " if the cluster is using the local Docker daemon.");
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks mostly good except for two micro details.

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java]: This was already changed in one of the commits

Comment (geoand) [reply extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java]: Oh sorry, I just saw what you meant.

**Comment by geoand** [comment]
Fixed

**Review by gsmet — APPROVED** [review]

**Pull request merged**